### PR TITLE
Support source_filters for Kibana index pattern.

### PR DESCRIPTION
**Pull request by simitt**
Support adding `SourceFilters` to Kibana Index Pattern.

Explanation for Kibana source filtering (from the Kibana Index Pattern / Source filtering section):
> Source filters can be used to exclude one or more fields when fetching the document source. This happens when viewing a document in the Discover app, or with a table displaying results from a saved search in the Dashboard app.

### CODE REVIEW

File: libbeat/kibana/source_filters.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	SourceFilters []string `config:"source_filters"`
+}
+
+func transformSourceFiltersFromYaml(yamlFile string) ([]common.MapStr, error) {
+	entries := []fieldInfo{}
+	cfg, err := yaml.NewConfigWithFile(yamlFile)
+	if err != nil {
+		return nil, err
+	}
+	cfg.Unpack(&entries)
```
Comment (exekias) [inline libbeat/kibana/source_filters.go]: unpack can return an error here

**Comment by simitt** [comment]
@ruflin regarding merging PR #5328 first - is it ready to merge? We need this feature, as otherwise Kibana discover section has issues handling APM data when uploading big sourcemaps.

File: libbeat/kibana/testdata/fields.yml [hunk 2]
```diff
@@ -1,5 +1,8 @@
 - key: test
   title: Test fields.yml
+  source_filters:
```
Comment (ruflin) [inline libbeat/kibana/testdata/fields.yml]: Can we name this `kibana.source_filters` as I think this is Kibana only? We discussed in the past to introduce a `kibana` specific namespace for the kibana only fields options. Lets start with the new additions for now and migrate the old ones step by step.

**Comment by ruflin** [comment]
@simitt Yes, PR is ready to go and now finally also Kibana tests should run again.

**Comment by ruflin** [comment]
@simitt Can you rebase this on top of https://github.com/elastic/beats/pull/5328 ?

**Comment by simitt** [comment]
@ruflin I added the `Kibana` namespace and did some naming refactoring for clarification.

File: libbeat/kibana/testdata/fields.yml [hunk 3]
```diff
@@ -1,5 +1,9 @@
 - key: test
   title: Test fields.yml
+  kibana:
```
Comment (ruflin) [inline libbeat/kibana/testdata/fields.yml]: Thanks for starting with the name spacing.

**Pull request merged**